### ISSUE TITLE
new versions do not start - error before start! (win10)

### ISSUE THREAD

**Issue by willi-spb** [dbeaver/dbeaver#36168]
### Description

![image](https://github.com/user-attachments/assets/fcb57df4-c54e-4b31-abdb-96080f54ef1c)
 
When updating - I get an error at startup - I can't launch. Old versions work normally.  

[1730869419415.log](https://github.com/user-attachments/files/17641880/1730869419415.log)

### DBeaver Version

24.2.4.0

### Operating System

Windows 10

**Comment by willi-spb** [comment]
I tried installing older versions -- 24.2.0 starts normally - all versions above (24.2.1. -24.2.4) do not start with an error (message - log)

**Comment by petr-hybler-ofti** [comment]
Ubuntu 24.04.1 LTS
DBeaver 24.2.4.202411031831


still broken ..

```
$ dbeaver-ce 
Picked up _JAVA_OPTIONS: -Duser.home=/home/<user>/snap/dbeaver-ce/330 -Dorg.eclipse.swt.browser.DefaultType=webkit -Dorg.eclipse.swt.browser.UseWebKitGTK=true -Ddbeaver.io.tmpdir=/home/<user>/Downloads
> Start Eclipse Jobs Mechanism [org.eclipse.core.jobs 3.15.400.v20240619-0602]
> Start Equinox Java Authentication and Authorization Service (JAAS) [org.eclipse.equinox.security 1.4.400.v20240702-1702]
> Start DBeaver Application Standalone [org.jkiss.dbeaver.ui.app.standalone 24.2.4.202411031831]
> Start Eclipse IDE UI Application [org.eclipse.ui.ide.application 1.5.500.v20240711-0817]
SWT OS.java Error: Failed to load swt-pi3, loading swt-pi4 as fallback.
Dbeaver:
An error has occurred. See the log file
/home/<user>/snap/dbeaver-ce/330/.local/share/DBeaverData/install-data/org.jkiss.dbeaver.product/24.2.4-202411031831/1730871809561.log.
```


```
$ cat /home/<user>/snap/dbeaver-ce/330/.local/share/DBeaverData/install-data/org.jkiss.dbeaver.product/24.2.4-202411031831/1730871809561.log
!SESSION 2024-11-06 06:43:29.300 -----------------------------------------------
eclipse.buildId=unknown
java.version=17.0.6
java.vendor=Eclipse Adoptium
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en_US
Framework arguments:  -eclipse.keyring /home/<user>/snap/dbeaver-ce/330/.local/share/DBeaverData/secure/secure_storage
Command-line arguments:  -os linux -ws gtk -arch x86_64
!ENTRY org.eclipse.osgi 4 0 2024-11-06 06:43:30.387
!MESSAGE An error occurred while automatically activating bundle org.eclipse.ui.ide (163).
!STACK 0
org.osgi.framework.BundleException: Exception in org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.start() of bundle org.eclipse.ui.ide.
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:854)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:775)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:1057)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:387)
	at org.eclipse.osgi.container.Module.doStart(Module.java:639)
	at org.eclipse.osgi.container.Module.start(Module.java:498)
	at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:528)
	at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:122)
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:620)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:348)
	at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:414)
	at org.eclipse.osgi.internal.loader.sources.SingleSourcePackage.loadClass(SingleSourcePackage.java:41)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:516)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:434)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:174)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.getRecentWorkspaces(DBeaverApplication.java:377)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.setIDEWorkspace(DBeaverApplication.java:353)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:192)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.invokeFramework(DBeaverLauncher.java:650)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.basicRun(DBeaverLauncher.java:610)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.run(DBeaverLauncher.java:1461)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.main(DBeaverLauncher.java:1434)
Caused by: java.lang.UnsatisfiedLinkError: Could not load SWT library. Reasons: 
	no swt-pi4-gtk-4966r5 in java.library.path: /var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void:/snap/dbeaver-ce/330/lib:/snap/dbeaver-ce/330/usr/lib:/snap/dbeaver-ce/330/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib:/snap/dbeaver-ce/330/gnome-platform/lib:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/dri:/var/lib/snapd/lib/gl:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/libunity:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/pulseaudio:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	no swt-pi4-gtk in java.library.path: /var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void:/snap/dbeaver-ce/330/lib:/snap/dbeaver-ce/330/usr/lib:/snap/dbeaver-ce/330/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib:/snap/dbeaver-ce/330/gnome-platform/lib:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/dri:/var/lib/snapd/lib/gl:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/libunity:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/pulseaudio:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	no swt-pi4 in java.library.path: /var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void:/snap/dbeaver-ce/330/lib:/snap/dbeaver-ce/330/usr/lib:/snap/dbeaver-ce/330/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib:/snap/dbeaver-ce/330/gnome-platform/lib:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/dri:/var/lib/snapd/lib/gl:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/libunity:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/pulseaudio:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	Can't load library: /home/<user>/snap/dbeaver-ce/330/.swt/lib/linux/x86_64/libswt-pi4-gtk-4966r5.so
	Can't load library: /home/<user>/snap/dbeaver-ce/330/.swt/lib/linux/x86_64/libswt-pi4-gtk.so
	Can't load library: /home/<user>/snap/dbeaver-ce/330/.swt/lib/linux/x86_64/libswt-pi4.so
	at org.eclipse.swt.internal.Library.loadLibrary(Library.java:345)
	at org.eclipse.swt.internal.Library.loadLibrary(Library.java:254)
	at org.eclipse.swt.internal.gtk.OS.<clinit>(OS.java:97)
	at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:209)
	at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:155)
	at org.eclipse.swt.widgets.Display.<clinit>(Display.java:170)
	at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.createProblemsViews(IDEWorkbenchPlugin.java:390)
	at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.start(IDEWorkbenchPlugin.java:347)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:833)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:825)
	... 31 more
Root exception:
java.lang.UnsatisfiedLinkError: Could not load SWT library. Reasons: 
	no swt-pi4-gtk-4966r5 in java.library.path: /var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void:/snap/dbeaver-ce/330/lib:/snap/dbeaver-ce/330/usr/lib:/snap/dbeaver-ce/330/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib:/snap/dbeaver-ce/330/gnome-platform/lib:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/dri:/var/lib/snapd/lib/gl:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/libunity:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/pulseaudio:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	no swt-pi4-gtk in java.library.path: /var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void:/snap/dbeaver-ce/330/lib:/snap/dbeaver-ce/330/usr/lib:/snap/dbeaver-ce/330/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib:/snap/dbeaver-ce/330/gnome-platform/lib:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/dri:/var/lib/snapd/lib/gl:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/libunity:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/pulseaudio:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	no swt-pi4 in java.library.path: /var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void:/snap/dbeaver-ce/330/lib:/snap/dbeaver-ce/330/usr/lib:/snap/dbeaver-ce/330/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib:/snap/dbeaver-ce/330/gnome-platform/lib:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/dri:/var/lib/snapd/lib/gl:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/libunity:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/pulseaudio:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	Can't load library: /home/<user>/snap/dbeaver-ce/330/.swt/lib/linux/x86_64/libswt-pi4-gtk-4966r5.so
	Can't load library: /home/<user>/snap/dbeaver-ce/330/.swt/lib/linux/x86_64/libswt-pi4-gtk.so
	Can't load library: /home/<user>/snap/dbeaver-ce/330/.swt/lib/linux/x86_64/libswt-pi4.so
	at org.eclipse.swt.internal.Library.loadLibrary(Library.java:345)
	at org.eclipse.swt.internal.Library.loadLibrary(Library.java:254)
	at org.eclipse.swt.internal.gtk.OS.<clinit>(OS.java:97)
	at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:209)
	at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:155)
	at org.eclipse.swt.widgets.Display.<clinit>(Display.java:170)
	at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.createProblemsViews(IDEWorkbenchPlugin.java:390)
	at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.start(IDEWorkbenchPlugin.java:347)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:833)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:825)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:775)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:1057)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:387)
	at org.eclipse.osgi.container.Module.doStart(Module.java:639)
	at org.eclipse.osgi.container.Module.start(Module.java:498)
	at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:528)
	at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:122)
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:620)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:348)
	at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:414)
	at org.eclipse.osgi.internal.loader.sources.SingleSourcePackage.loadClass(SingleSourcePackage.java:41)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:516)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:434)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:174)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.getRecentWorkspaces(DBeaverApplication.java:377)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.setIDEWorkspace(DBeaverApplication.java:353)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:192)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.invokeFramework(DBeaverLauncher.java:650)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.basicRun(DBeaverLauncher.java:610)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.run(DBeaverLauncher.java:1461)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.main(DBeaverLauncher.java:1434)
!ENTRY org.eclipse.osgi 4 0 2024-11-06 06:43:30.390
!MESSAGE Application error
!STACK 1
java.lang.NoClassDefFoundError: org/eclipse/ui/internal/ide/ChooseWorkspaceData
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.getRecentWorkspaces(DBeaverApplication.java:377)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.setIDEWorkspace(DBeaverApplication.java:353)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:192)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.invokeFramework(DBeaverLauncher.java:650)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.basicRun(DBeaverLauncher.java:610)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.run(DBeaverLauncher.java:1461)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.main(DBeaverLauncher.java:1434)
Caused by: java.lang.ClassNotFoundException: An error occurred while automatically activating bundle org.eclipse.ui.ide (163).
	at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:134)
	at org.eclipse.osgi.internal.loader.classpath.ClasspathManager.findLocalClass(ClasspathManager.java:620)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.findLocalClass(ModuleClassLoader.java:348)
	at org.eclipse.osgi.internal.loader.BundleLoader.findLocalClass(BundleLoader.java:414)
	at org.eclipse.osgi.internal.loader.sources.SingleSourcePackage.loadClass(SingleSourcePackage.java:41)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass0(BundleLoader.java:516)
	at org.eclipse.osgi.internal.loader.BundleLoader.findClass(BundleLoader.java:434)
	at org.eclipse.osgi.internal.loader.ModuleClassLoader.loadClass(ModuleClassLoader.java:174)
	at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
	... 16 more
Caused by: org.osgi.framework.BundleException: Exception in org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.start() of bundle org.eclipse.ui.ide.
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:854)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.start(BundleContextImpl.java:775)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.startWorker0(EquinoxBundle.java:1057)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$EquinoxModule.startWorker(EquinoxBundle.java:387)
	at org.eclipse.osgi.container.Module.doStart(Module.java:639)
	at org.eclipse.osgi.container.Module.start(Module.java:498)
	at org.eclipse.osgi.framework.util.SecureAction.start(SecureAction.java:528)
	at org.eclipse.osgi.internal.hooks.EclipseLazyStarter.postFindLocalClass(EclipseLazyStarter.java:122)
	... 24 more
Caused by: java.lang.UnsatisfiedLinkError: Could not load SWT library. Reasons: 
	no swt-pi4-gtk-4966r5 in java.library.path: /var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void:/snap/dbeaver-ce/330/lib:/snap/dbeaver-ce/330/usr/lib:/snap/dbeaver-ce/330/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib:/snap/dbeaver-ce/330/gnome-platform/lib:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/dri:/var/lib/snapd/lib/gl:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/libunity:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/pulseaudio:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	no swt-pi4-gtk in java.library.path: /var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void:/snap/dbeaver-ce/330/lib:/snap/dbeaver-ce/330/usr/lib:/snap/dbeaver-ce/330/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib:/snap/dbeaver-ce/330/gnome-platform/lib:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/dri:/var/lib/snapd/lib/gl:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/libunity:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/pulseaudio:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	no swt-pi4 in java.library.path: /var/lib/snapd/lib/gl:/var/lib/snapd/lib/gl32:/var/lib/snapd/void:/snap/dbeaver-ce/330/lib:/snap/dbeaver-ce/330/usr/lib:/snap/dbeaver-ce/330/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu:/snap/dbeaver-ce/330/gnome-platform/usr/lib:/snap/dbeaver-ce/330/gnome-platform/lib:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/dri:/var/lib/snapd/lib/gl:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/libunity:/snap/dbeaver-ce/330/gnome-platform/usr/lib/x86_64-linux-gnu/pulseaudio:/usr/java/packages/lib:/usr/lib64:/lib64:/lib:/usr/lib
	Can't load library: /home/<user>/snap/dbeaver-ce/330/.swt/lib/linux/x86_64/libswt-pi4-gtk-4966r5.so
	Can't load library: /home/<user>/snap/dbeaver-ce/330/.swt/lib/linux/x86_64/libswt-pi4-gtk.so
	Can't load library: /home/<user>/snap/dbeaver-ce/330/.swt/lib/linux/x86_64/libswt-pi4.so
	at org.eclipse.swt.internal.Library.loadLibrary(Library.java:345)
	at org.eclipse.swt.internal.Library.loadLibrary(Library.java:254)
	at org.eclipse.swt.internal.gtk.OS.<clinit>(OS.java:97)
	at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:209)
	at org.eclipse.swt.internal.Converter.wcsToMbcs(Converter.java:155)
	at org.eclipse.swt.widgets.Display.<clinit>(Display.java:170)
	at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.createProblemsViews(IDEWorkbenchPlugin.java:390)
	at org.eclipse.ui.internal.ide.IDEWorkbenchPlugin.start(IDEWorkbenchPlugin.java:347)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:833)
	at org.eclipse.osgi.internal.framework.BundleContextImpl$2.run(BundleContextImpl.java:1)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.startActivator(BundleContextImpl.java:825)
	... 31 more
```

**Comment by petr-hybler-ofti** [comment]
tho this still works as a workaround (keeps the configuration intact)

https://github.com/dbeaver/dbeaver/issues/36006#issuecomment-2428527154

**Comment by serge-rider** [comment]
@petr-hybler-ofti In 24.2.1 we migrated to the latest Eclipse platform, this may be an issue.

For Linux it may make sense to update GTK version:
https://stackoverflow.com/questions/76956123/could-not-load-swt-library-error-no-swt-pi4-in-in-java-library-path

**Comment by serge-rider** [comment]
@willi-spb This is weird. But may be a result of recent issues in Eclipse SWT with older Windows versions support. Seems to be a duplicate of https://github.com/dbeaver/dbeaver/issues/35694 .

Could you try to download EA version (https://dbeaver.io/files/ea) to verify the latest fix?

**Comment by petr-hybler-ofti** [comment]
> @petr-hybler-ofti In 24.2.1 we migrated to the latest Eclipse platform, this may be an issue.
> 
> For Linux it may make sense to update GTK version: https://stackoverflow.com/questions/76956123/could-not-load-swt-library-error-no-swt-pi4-in-in-java-library-path
```
$ dpkg --get-selections | grep libgtk
libgtk-3-0t64:amd64				install
libgtk-3-bin					install
libgtk-3-common					install
libgtk-4-1:amd64				install
libgtk-4-bin					install
libgtk-4-common					install
libgtk-4-media-gstreamer			install
libgtk2.0-0t64:amd64				install
libgtk2.0-bin					install
libgtk2.0-common				install
libgtk3-perl					install
libgtkmm-3.0-1t64:amd64				install
libgtkmm-4.0-0:amd64				install
libgtksourceview-4-0:amd64			install
libgtksourceview-4-common			install
libgtksourceview-5-0:amd64			install
libgtksourceview-5-common			install
```

I don't see anything else available

**Comment by serge-rider** [comment]
@petr-hybler-ofti it looks like an issue with our latest snap image build (similar to https://github.com/dbeaver/dbeaver/issues/19779). Investigating

**Comment by ShadelessFox** [comment]
@willi-spb can you please name the exact version of your OS using the `winver` command? You are using release versions before 24.2.5, not the early access versions, correct?

**Comment by willi-spb** [comment]
> @willi-spb This is weird. But may be a result of recent issues in Eclipse SWT with older Windows versions support. Seems to be a duplicate of #35694 .
> 
> Could you try to download EA version (https://dbeaver.io/files/ea) to verify the latest fix?
__________ 
  
Thank you - 24.2.5 started normally after installation. I haven't seen any errors yet.
